RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+
+# Utility Changes
+
+Amplify is proposing the following changes to our core utilities.
+
+## TypeScript support for Amplify Hub channels
+
+We are improving developer experience by adding strict type support to Hub channels, events, and payloads. An example of the developer experience when listening for `auth` events is highlighted below.
+
+**Amplify v5 & v6 (`aws-amplify@^5`)**
+
+```Typescript
+Hub.dispatch("auth", {
+  event: "signInFailure",
+  data: new AuthError("Sign-in failed"),
+});
+
+// Events and payload data are not inferred.
+Hub.listen(channel, ({ payload }) => {
+  switch (payload.event) {
+    case "signInFailure":
+      const data = payload.data;
+      break;
+  }
+});
+```
+
+**Amplify v5 DX**
+
+https://user-images.githubusercontent.com/70438514/222475294-9cf9f535-b717-4725-9e5a-7c362e74a054.mov
+
+<br />
+
+**Proposed Amplify v6 DX**
+
+https://user-images.githubusercontent.com/70438514/222512918-ea72b2c4-9568-45e6-86c6-3c5226820e7d.mov
+
+## TypeScript support for custom Hub channels
+
+We are improving developer experience by adding strict type support to custom Hub channels, events, and payloads.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+const customChannel = "custom_channel";
+const customEvent = "custom_event";
+const customData = "custom_data";
+
+Hub.dispatch(channel, {
+  event: customEvent,
+  data: customData
+});
+
+Hub.listen(channel, ({ payload }) => {
+  switch (payload.event) {
+    case customEvent:
+      const data = payload.data;
+      break;
+  }
+});
+```
+
+**Amplify v5 DX**
+
+https://user-images.githubusercontent.com/70438514/222475304-128de96a-9a06-4fcc-8f0d-8f19592ae9eb.mov
+
+<br />
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+type customChannel = "custom_channel";
+
+// Each key in the map represents a payload event and the key value is the data type for that event.
+// Note: If an event is assigned the null type, the payload object will not contain a data key.
+type customEventDataMap = {
+  A: number;
+  B: string;
+  C: null;
+  D: Object
+};
+
+Hub.dispatch<customChannel, customEventDataMap >("custom_channel", {
+  event: "A",
+  data: 42
+});
+
+Hub.listen<customChannel, customEventDataMap>("custom_channel", ({payload}) => {
+  switch(payload.event) {
+    case "A":
+      payload.data;
+      break;
+    case "B":
+      payload.data;
+      break;
+    case "C":
+      payload.data;
+      break;
+    case "D":
+      payload.data;
+      break;
+  }
+});
+```
+
+**Proposed Amplify v6 DX**
+
+https://user-images.githubusercontent.com/70438514/222475311-36b4f8bc-d338-4ad1-9910-5d3518fb71c0.mov
+
+## TypeScript support for Amplify Configuration
+
+To help customers configure Amplify categories, we are introducing type support for the `Amplify.configure` API. The examples below demonstrate an `Auth` configuration.
```

Comment:
Can we add the value here. "This allows you to easily setup your AWS resources if you are connecting Amplify JS to resources you have not created directly with the Amplify CLI." Something along these lines